packages/theme/src/types.ts

```diff
@@ -23,6 +23,15 @@ export interface ThemeProviderSettings {
 		 */
 		bg?: string;
 	};
+
+	/**
+	 * The density of the theme. If left unspecified, the theme inherits from
+	 * the density of the closest `ThemeProvider`, or uses the default density
+	 * if there is no inherited density.
+	 *
+	 * @default undefined
+	 */
+	density?: undefined | 'default' | 'compact' | 'comfortable';
```

Comment:
Just a thought, no strong opinion, but would something like this be more intuitive? Personally I think it would be easier to understand when skimming the docs, since it's closer to the CSS mental model.

```suggestion
	/**
	 * The density of the theme. If left unspecified, the theme inherits from
	 * the density of the closest `ThemeProvider`, or uses the default density
	 * if there is no inherited density.
	 *
	 * @default 'inherit'
	 */
	density?: 'inherit' | 'default' | 'compact' | 'comfortable';
```

Reply:
Hm, I think there's some advantages to that, particularly around explicitness. However, on the whole, I don't know that it's a change I want to make now:

- The idea of using `undefined` (absence of value) as interpreted to inherit from the parent is consistent with how we're treating color customization above ([source](https://github.com/WordPress/gutenberg/blob/01136d19b0985d12ff42c3cb7329ea421a583d28/packages/theme/src/types.ts#L11-L24))
- While not really a blocker in itself, doing this incurs some internal technical complexity because we need to normalize the `'inherit'` value _back to_ `undefined` in order to effect the inherit behavior by ensuring that `data-wpds-density` isn't assigned ([source](https://github.com/WordPress/gutenberg/blob/56672d58dab126c9e9a1c3cd5a6d1cc48c5f9df6/packages/theme/src/theme-provider.tsx#L80)) because, unlike `[data-wpds-density="compact"]`, etc., we don't have an explicit selector for the inherited behavior (there is no `[data-wpds-density="inherit"]`).
- The suggestion feels like a logical step in part because of how I've gone out of my way to be explicit here, though in retrospect I'm not fully convinced it was necessary to be so explicit (i.e. including `@default undefined` and including `undefined` in the type union for `density`).

Reply:
That's fair. Maybe it looks weird to me because all our "physical" components do `@default 'default'`, and this is the first time we're doing props with inheritance.